runtime/src/bank.rs

```diff
@@ -331,33 +331,49 @@ impl CachedExecutors {
         })
     }
 
-    fn put(&mut self, pubkey: &Pubkey, executor: Arc<dyn Executor>) {
-        let entry = if let Some(mut entry) = self.executors.remove(pubkey) {
-            entry.executor = executor;
-            entry
-        } else {
+    fn put(&mut self, executors: &[(&Pubkey, Arc<dyn Executor>)]) {
+        let new_executors: Option<Vec<_>> = executors
+            .iter()
+            .map(|(key, executor)| {
+                if let Some(mut entry) = self.executors.remove(key) {
+                    entry.executor = executor.clone();
+                    let _ = self.executors.insert(**key, entry);
+                    None
+                } else {
+                    Some((key, executor))
+                }
+            })
+            .collect();
+
+        if let Some(mut new_executors) = new_executors {
             if self.executors.len() >= self.max {
-                let mut least = u64::MAX;
-                let default_key = Pubkey::default();
-                let mut least_key = &default_key;
-
-                for (key, entry) in self.executors.iter() {
-                    let count = entry.prev_epoch_count + entry.epoch_count.load(Relaxed);
-                    if count < least {
-                        least = count;
-                        least_key = key;
+                for _ in 0..new_executors.len() {
+                    if self.executors.len() >= self.max {
+                        let mut least = u64::MAX;
+                        let default_key = Pubkey::default();
+                        let mut least_key = &default_key;
+
+                        for (key, entry) in self.executors.iter() {
+                            let count = entry.prev_epoch_count + entry.epoch_count.load(Relaxed);
+                            if count < least {
+                                least = count;
+                                least_key = key;
+                            }
+                        }
+                        let least_key = *least_key;
+                        let _ = self.executors.remove(&least_key);
                     }
                 }
-                let least_key = *least_key;
-                let _ = self.executors.remove(&least_key);
             }
-            CachedExecutorsEntry {
-                prev_epoch_count: 0,
-                epoch_count: AtomicU64::new(0),
-                executor,
+            for (key, executor) in new_executors.drain(..) {
+                let entry = CachedExecutorsEntry {
+                    prev_epoch_count: 0,
+                    epoch_count: AtomicU64::new(0),
```

Comment:
I'll think about how this'll need modified for the random primer counts

Reply:
I had:
```
epoch_count: AtomicU64::new(thread_rng().gen_range(0, paths.len())
```
Even something that simple worked pretty good

Reply:
That'll work worse the longer the cache runs though, no?

Reply:
It only helps the surface thrash of the last few additions

Reply:
> It only helps the surface thrash of the last few additions

Yeah exactly.  With the rand count of the 85-percentile entry, new entries can get hucked further into the cache as entries age

Reply:
as long as we don't huck in 1 offs that deep enough that they never leave.  Also a reason for adding better decay